### PR TITLE
Update ADU device props to GA fields

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

File: sdk/tests/iot/adu/test_az_iot_adu.c [hunk 1]
```diff
@@ -29,14 +29,16 @@
 
 static uint8_t expected_agent_state_payload[]
```
Comment (vaavva) [inline sdk/tests/iot/adu/test_az_iot_adu.c:30]: Do we need to update the manifest version in these tests as well for this change?

Comment (danewalton) [reply sdk/tests/iot/adu/test_az_iot_adu.c:30]: Good question. We can. The tests are just parsing out a number so it realistically could be `10000` and it wouldn't matter. As long as the format of the payload is correct.

**Comment by danewalton** [comment]
Verified with @ewertons that the only other change is the manifest version going from 4 to 5. Updated it in the unit tests.

**Review by vaavva — APPROVED** [review]

**Review by ewertons — COMMENTED** [review]
Approved

File: sdk/src/azure/iot/az_iot_adu_client.c [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
+#define AZ_IOT_ADU_CLIENT_AGENT_CONTRACT_MODEL_ID "dtmi:azure:iot:deviceUpdateContractModel;2"
 
 /* Define the ADU agent property name "agent" and sub property names.  */
 #define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_AGENT "agent"
 
 #define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_DEVICEPROPERTIES "deviceProperties"
 #define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_MANUFACTURER "manufacturer"
 #define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_MODEL "model"
-#define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_INTERFACE_ID "interfaceId"
+#define AZ_IOT_ADU_CLIENT_AGENT_PROPERTY_NAME_CONTRACT_MODEL_ID "contractModelId"
```
Comment (ahsonkhan) [inline sdk/src/azure/iot/az_iot_adu_client.c:23]: Add a bug fix Changelog entry?

Comment (danewalton) [reply sdk/src/azure/iot/az_iot_adu_client.c:23]: done
https://github.com/Azure/azure-sdk-for-c/pull/2451

**Pull request merged**